### ISSUE TITLE
Dynatrace jenkins plugin not connecting, creating stacktrace.

### ISSUE THREAD

**Issue by Harriebo** [jenkinsci/dynatrace-plugin#9]
When configuring our dynatrace server in the plugin it says unable to connect. It says:

Connection failure - System profile with the given name can not be found.

This while I do not have a option to specify a profile...
I thinks its trying to acces the pofile list:   /api/v1/profiles
But uses the wrong url.

The jenkins log says:
line 1:0 no viable alternative at character '<'
line 1:1 no viable alternative at character 'h'
line 1:3 mismatched character 'm' expecting 'r'
line 1:4 no viable alternative at character 'l'
line 1:5 no viable alternative at character '>'
line 2:0 no viable alternative at character '<'
line 2:1 no viable alternative at character 'h'
line 2:2 no viable alternative at character 'e'
line 2:3 no viable alternative at character 'a'
line 2:4 no viable alternative at character 'd'
line 2:5 no viable alternative at character '>'
line 3:0 no viable alternative at character '<'
line 3:1 no viable alternative at character 'm'
line 3:2 no viable alternative at character 'e'
line 3:4 mismatched character 'a' expecting 'r'
line 3:6 no viable alternative at character 'h'
line 3:8 mismatched character 't' expecting 'r'
line 3:9 no viable alternative at character 'p'
line 3:11 required (...)+ loop did not match anything at character 'e'
line 3:11 no viable alternative at character 'e'
line 3:12 no viable alternative at character 'q'
line 3:13 no viable alternative at character 'u'
line 3:14 no viable alternative at character 'i'
line 3:15 no viable alternative at character 'v'
line 3:16 no viable alternative at character '='
line 3:32 no viable alternative at character 'c'
line 3:33 no viable alternative at character 'o'
line 3:35 mismatched character 't' expecting 'u'
line 3:36 no viable alternative at character 'e'
line 3:38 mismatched character 't' expecting 'u'
line 3:39 no viable alternative at character '='
line 3:70 no viable alternative at character '/'
line 3:71 no viable alternative at character '>'
line 4:0 no viable alternative at character '<'
line 4:2 mismatched character 'i' expecting 'r'
line 4:4 mismatched character 'l' expecting 'r'
line 4:5 no viable alternative at character 'e'
line 4:6 no viable alternative at character '>'
line 4:7 no viable alternative at character 'E'
line 4:8 no viable alternative at character 'r'
line 4:9 no viable alternative at character 'r'
line 4:10 no viable alternative at character 'o'
line 4:11 no viable alternative at character 'r'
line 4:17 no viable alternative at character '<'
line 4:18 no viable alternative at character '/'
line 4:20 mismatched character 'i' expecting 'r'
line 4:22 mismatched character 'l' expecting 'r'
line 4:23 no viable alternative at character 'e'
line 4:24 no viable alternative at character '>'
line 5:0 no viable alternative at character '<'
line 5:1 no viable alternative at character '/'
line 5:2 no viable alternative at character 'h'
line 5:3 no viable alternative at character 'e'
line 5:4 no viable alternative at character 'a'
line 5:5 no viable alternative at character 'd'
line 5:6 no viable alternative at character '>'
line 6:0 no viable alternative at character '<'
line 6:1 no viable alternative at character 'b'
line 6:2 no viable alternative at character 'o'
line 6:3 no viable alternative at character 'd'
line 6:4 no viable alternative at character 'y'
line 6:5 no viable alternative at character '>'
line 7:0 no viable alternative at character '<'
line 7:1 no viable alternative at character 'h'
line 7:3 no viable alternative at character '>'
line 7:4 no viable alternative at character 'H'
line 7:5 no viable alternative at character 'T'
line 7:6 no viable alternative at character 'T'
line 7:7 no viable alternative at character 'P'
line 7:9 no viable alternative at character 'E'
line 7:10 no viable alternative at character 'R'
line 7:11 no viable alternative at character 'R'
line 7:12 no viable alternative at character 'O'
line 7:13 no viable alternative at character 'R'
line 7:19 no viable alternative at character '<'
line 7:20 no viable alternative at character '/'
line 7:21 no viable alternative at character 'h'
line 7:23 no viable alternative at character '>'
line 8:0 no viable alternative at character '<'
line 8:1 no viable alternative at character 'p'
line 8:2 no viable alternative at character '>'
line 8:3 no viable alternative at character 'P'
line 8:4 no viable alternative at character 'r'
line 8:5 no viable alternative at character 'o'
line 8:6 no viable alternative at character 'b'
line 8:7 no viable alternative at character 'l'
line 8:8 no viable alternative at character 'e'
line 8:9 no viable alternative at character 'm'
line 8:11 no viable alternative at character 'a'
line 8:12 no viable alternative at character 'c'
line 8:13 no viable alternative at character 'c'
line 8:14 no viable alternative at character 'e'
line 8:15 no viable alternative at character 's'
line 8:16 no viable alternative at character 's'
line 8:17 no viable alternative at character 'i'
line 8:19 mismatched character 'g' expecting 'u'
line 8:21 no viable alternative at character '/'
line 8:22 no viable alternative at character 'a'
line 8:23 no viable alternative at character 'p'
line 8:24 no viable alternative at character 'i'
line 8:25 no viable alternative at character '/'
line 8:26 no viable alternative at character 'v'
line 8:28 no viable alternative at character '/'
line 8:29 no viable alternative at character 'p'
line 8:30 no viable alternative at character 'r'
line 8:31 no viable alternative at character 'o'
line 8:33 mismatched character 'i' expecting 'a'
line 8:34 no viable alternative at character 'l'
line 8:35 no viable alternative at character 'e'
line 8:36 no viable alternative at character 's'
line 8:37 no viable alternative at character '/'
line 8:38 no viable alternative at character '.'
line 8:40 no viable alternative at character 'R'
line 8:41 no viable alternative at character 'e'
line 8:42 no viable alternative at character 'a'
line 8:43 no viable alternative at character 's'
line 8:44 no viable alternative at character 'o'
line 8:46 mismatched character ':' expecting 'u'
line 9:0 no viable alternative at character '<'
line 9:1 no viable alternative at character 'p'
line 9:2 no viable alternative at character 'r'
line 9:3 no viable alternative at character 'e'
line 9:4 no viable alternative at character '>'
line 9:9 no viable alternative at character 'N'
line 9:10 no viable alternative at character 'o'
line 9:12 mismatched character ' ' expecting 'r'
line 9:13 no viable alternative at character 'F'
line 9:14 no viable alternative at character 'o'
line 9:15 no viable alternative at character 'u'
line 9:17 mismatched character 'd' expecting 'u'
line 9:18 no viable alternative at character '<'
line 9:19 no viable alternative at character '/'
line 9:20 no viable alternative at character 'p'
line 9:21 no viable alternative at character 'r'
line 9:22 no viable alternative at character 'e'
line 9:23 no viable alternative at character '>'
line 9:24 no viable alternative at character '<'
line 9:25 no viable alternative at character '/'
line 9:26 no viable alternative at character 'p'
line 9:27 no viable alternative at character '>'
line 10:0 no viable alternative at character '<'
line 10:1 no viable alternative at character 'h'
line 10:2 no viable alternative at character 'r'
line 10:4 no viable alternative at character '/'
line 10:5 no viable alternative at character '>'
line 10:6 no viable alternative at character '<'
line 10:7 no viable alternative at character 'i'
line 10:8 no viable alternative at character '>'
line 10:9 no viable alternative at character '<'
line 10:10 no viable alternative at character 's'
line 10:11 no viable alternative at character 'm'
line 10:12 no viable alternative at character 'a'
line 10:13 no viable alternative at character 'l'
line 10:14 no viable alternative at character 'l'
line 10:15 no viable alternative at character '>'
line 10:16 no viable alternative at character 'P'
line 10:17 no viable alternative at character 'o'
line 10:18 no viable alternative at character 'w'
line 10:19 no viable alternative at character 'e'
line 10:20 no viable alternative at character 'r'
line 10:21 no viable alternative at character 'e'
line 10:22 no viable alternative at character 'd'
line 10:24 no viable alternative at character 'b'
line 10:25 no viable alternative at character 'y'
line 10:27 no viable alternative at character 'J'
line 10:28 no viable alternative at character 'e'
line 10:30 mismatched character 't' expecting 'r'
line 10:31 no viable alternative at character 'y'
line 10:33 no viable alternative at character '/'
line 10:34 no viable alternative at character '/'
line 10:35 no viable alternative at character '<'
line 10:36 no viable alternative at character '/'
line 10:37 no viable alternative at character 's'
line 10:38 no viable alternative at character 'm'
line 10:39 no viable alternative at character 'a'
line 10:40 no viable alternative at character 'l'
line 10:41 no viable alternative at character 'l'
line 10:42 no viable alternative at character '>'
line 10:43 no viable alternative at character '<'
line 10:44 no viable alternative at character '/'
line 10:45 no viable alternative at character 'i'
line 10:46 no viable alternative at character '>'
line 11:0 no viable alternative at character '<'
line 11:1 no viable alternative at character '/'
line 11:2 no viable alternative at character 'b'
line 11:3 no viable alternative at character 'o'
line 11:4 no viable alternative at character 'd'
line 11:5 no viable alternative at character 'y'
line 11:6 no viable alternative at character '>'
line 12:0 no viable alternative at character '<'
line 12:1 no viable alternative at character '/'
line 12:2 no viable alternative at character 'h'
line 12:4 mismatched character 'm' expecting 'r'
line 12:5 no viable alternative at character 'l'
line 12:6 no viable alternative at character '>'
line 3:17 no viable alternative at input '"Content-Type"'

com.dynatrace.sdk.server.exceptions.ServerResponseException: Not Found
at com.dynatrace.sdk.server.Service.doRequest(Service.java:147)
at com.dynatrace.sdk.server.Service.doPostRequest(Service.java:202)
at com.dynatrace.sdk.server.Service.doPostRequest(Service.java:207)
at com.dynatrace.sdk.server.Service.doPostRequest(Service.java:220)
at com.dynatrace.sdk.server.sessions.Sessions.startRecording(Sessions.java:61)
at com.dynatrace.jenkins.dashboard.TABuildWrapper.setUp(TABuildWrapper.java:160)
at org.jenkinsci.plugins.workflow.steps.CoreWrapperStep$Execution.start(CoreWrapperStep.java:80)
at org.jenkinsci.plugins.workflow.cps.DSL.invokeStep(DSL.java:224)
at org.jenkinsci.plugins.workflow.cps.DSL.invokeDescribable(DSL.java:353)
at org.jenkinsci.plugins.workflow.cps.DSL.invokeMethod(DSL.java:153)
at org.jenkinsci.plugins.workflow.cps.CpsScript.invokeMethod(CpsScript.java:108)
at sun.reflect.GeneratedMethodAccessor175.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:498)
at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1215)
at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1024)
at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:42)
at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
at org.kohsuke.groovy.sandbox.impl.Checker$1.call(Checker.java:155)
at org.kohsuke.groovy.sandbox.GroovyInterceptor.onMethodCall(GroovyInterceptor.java:23)
at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.SandboxInterceptor.onMethodCall(SandboxInterceptor.java:133)
at org.kohsuke.groovy.sandbox.impl.Checker$1.call(Checker.java:153)
at org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:157)
at org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:127)
at org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:127)
at com.cloudbees.groovy.cps.sandbox.SandboxInvoker.methodCall(SandboxInvoker.java:17)
at com.cloudbees.groovy.cps.impl.ContinuationGroup.methodCall(ContinuationGroup.java:57)
at com.cloudbees.groovy.cps.impl.FunctionCallBlock$ContinuationImpl.dispatchOrArg(FunctionCallBlock.java:109)
at com.cloudbees.groovy.cps.impl.FunctionCallBlock$ContinuationImpl.fixArg(FunctionCallBlock.java:82)
at sun.reflect.GeneratedMethodAccessor79.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:498)
at com.cloudbees.groovy.cps.impl.ContinuationPtr$ContinuationImpl.receive(ContinuationPtr.java:72)
at com.cloudbees.groovy.cps.impl.ClosureBlock.eval(ClosureBlock.java:46)
at com.cloudbees.groovy.cps.Next.step(Next.java:83)
at com.cloudbees.groovy.cps.Continuable$1.call(Continuable.java:174)
at com.cloudbees.groovy.cps.Continuable$1.call(Continuable.java:163)
at org.codehaus.groovy.runtime.GroovyCategorySupport$ThreadCategoryInfo.use(GroovyCategorySupport.java:122)

**Comment by Harriebo** [comment]
I forgot to add, if I try to access it my self from the jenkins server there is no problem:
wget https://urltodynatrace:8021/api/v1/profiles --username=xxx --pasword=xxx
works and returns a json list.

**Comment by dglugla** [comment]
Hi @Harriebo 

The stacktrace you provided was generated by build job, not connection verification.
TABuildWrapper.setUp method is executed when you click 'Test connection' on project configuration screen, and that's where you need to provide system profile's name.
Please verify if the stacktrace is valid and if you have all the server settings configured properly (Manage Jenkins -> Configure System -> Dynatrace Application Monitoring) and the server connection is verified.

BR

**Comment by Harriebo** [comment]
Hi,
I think you are right, I mixed up the configure output with the output
during the build. I managed to resolve the configuration issue by updating
all other plugins and it stopped complaining and it accepts te
configuration now as a succesful connection. So that is a good start, but I
still see stack traces during build and I can not get a test ID from the
Dynatrace server:

com.dynatrace.sdk.server.exceptions.ServerResponseException: No connection
to the Performance Warehouse. Please establish a connection from the
Dynatrace Server to the Performance Warehouse to see data in this view.
        at com.dynatrace.sdk.server.Service.doRequest(Service.java:147)
        at com.dynatrace.sdk.server.Service.doPostRequest(Service.java:202)
        at com.dynatrace.sdk.server.Service.doPostRequest(Service.java:207)
        at com.dynatrace.sdk.server.Service.doPostRequest(Service.java:220)
        at
com.dynatrace.sdk.server.testautomation.TestAutomation.createTestRun(TestAutomation.java:71)
        at
com.dynatrace.jenkins.dashboard.TATestRunRegistrationBuildStep.perform(TATestRunRegistrationBuildStep.java:115)
        at
com.dynatrace.jenkins.dashboard.steps.TATestRunRegistrationPipelineStep$Execution.run(TATestRunRegistrationPipelineStep.java:74)
        at
com.dynatrace.jenkins.dashboard.steps.TATestRunRegistrationPipelineStep$Execution.run(TATestRunRegistrationPipelineStep.java:48)
        at
org.jenkinsci.plugins.workflow.steps.AbstractSynchronousStepExecution.start(AbstractSynchronousStepExecution.java:42)
        at org.jenkinsci.plugins.workflow.cps.DSL.invokeStep(DSL.java:224)
        at org.jenkinsci.plugins.workflow.cps.DSL.invokeMethod(DSL.java:150)
        at
org.jenkinsci.plugins.workflow.cps.CpsScript.invokeMethod(CpsScript.java:108)
        at sun.reflect.GeneratedMethodAccessor4120.invoke(Unknown Source)
        at
sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at
org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
        at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1215)
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1024)
        at
org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:42)
        at
org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
        at
org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
        at org.kohsuke.groovy.sandbox.impl.Checker$1.call(Checker.java:155)
        at
org.kohsuke.groovy.sandbox.GroovyInterceptor.onMethodCall(GroovyInterceptor.java:23)
        at
org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.SandboxInterceptor.onMethodCall(SandboxInterceptor.java:133)
        at org.kohsuke.groovy.sandbox.impl.Checker$1.call(Checker.java:153)
        at
org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:157)
        at
org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:127)
        at
org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:127)
        at
org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:127)

com.dynatrace.sdk.server.exceptions.ServerResponseException: Session is
already recording
        at com.dynatrace.sdk.server.Service.doRequest(Service.java:147)
        at com.dynatrace.sdk.server.Service.doPostRequest(Service.java:202)
        at com.dynatrace.sdk.server.Service.doPostRequest(Service.java:207)
        at com.dynatrace.sdk.server.Service.doPostRequest(Service.java:220)
        at
com.dynatrace.sdk.server.sessions.Sessions.startRecording(Sessions.java:61)
        at
com.dynatrace.jenkins.dashboard.TABuildWrapper.setUp(TABuildWrapper.java:160)
        at
org.jenkinsci.plugins.workflow.steps.CoreWrapperStep$Execution.start(CoreWrapperStep.java:80)
        at org.jenkinsci.plugins.workflow.cps.DSL.invokeStep(DSL.java:224)
        at
org.jenkinsci.plugins.workflow.cps.DSL.invokeDescribable(DSL.java:353)
        at org.jenkinsci.plugins.workflow.cps.DSL.invokeMethod(DSL.java:153)
        at
org.jenkinsci.plugins.workflow.cps.CpsScript.invokeMethod(CpsScript.java:108)
        at sun.reflect.GeneratedMethodAccessor4120.invoke(Unknown Source)
        at
sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at
org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
        at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1215)
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1024)
        at
org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:42)
        at
org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
        at
org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
        at org.kohsuke.groovy.sandbox.impl.Checker$1.call(Checker.java:155)
        at
org.kohsuke.groovy.sandbox.GroovyInterceptor.onMethodCall(GroovyInterceptor.java:23)
        at
org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.SandboxInterceptor.onMethodCall(SandboxInterceptor.java:133)
        at org.kohsuke.groovy.sandbox.impl.Checker$1.call(Checker.java:153)
        at
org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:157)
        at
org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:127)
        at
org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:127)
        at
com.cloudbees.groovy.cps.sandbox.SandboxInvoker.methodCall(SandboxInvoker.java:17)
        at
com.cloudbees.groovy.cps.impl.ContinuationGroup.methodCall(ContinuationGroup.java:57)
        at
com.cloudbees.groovy.cps.impl.FunctionCallBlock$ContinuationImpl.dispatchOrArg(FunctionCallBlock.java:109)
        at
com.cloudbees.groovy.cps.impl.FunctionCallBlock$ContinuationImpl.fixArg(FunctionCallBlock.java:82)
        at sun.reflect.GeneratedMethodAccessor79.invoke(Unknown Source)

We have multiple times

appMonBuildEnvironment(systemProfile: 'Jenkins', recordSession:
dynatraceEnabled) {

}


in the jenkins file because we have a staged build with multiple test suits
on different nodes. So for each node we have the tag again.
The profile is there and the user has access to it for sure. Is this the
right setup, or should there be one tag for the whole jenkins file that
overs multiple nodes?
We do reuse the samde test id requested, but there is no valid response,
the build log says:

Registering test run via Dynatrace Server REST interface...
ERROR: Dynatrace AppMon Plugin - build step execution failed (see the
stacktrace to get more information):
com.dynatrace.sdk.server.exceptions.ServerResponseException: No
connection to the Performance Warehouse. Please establish a connection
from the Dynatrace Server to the Performance Warehouse to see data in
this view.

(twice because I try to get a unit test and a functional test id)

But during build it says things like:

[unit and integration tests] Starting session recording via Dynatrace
Server REST interface...
[unit and integration tests] Dynatrace session
https://dsacc.bva-tools.com:8021/api/v2/sessions/Jenkins%252F20171016144847_0.t0.session
has been started

and

[second-site-functional-suite] Starting session recording via
Dynatrace Server REST interface...
[second-site-functional-suite] ERROR: Dynatrace AppMon Plugin - build
set up failed (see the stacktrace to get more information):
[second-site-functional-suite]
com.dynatrace.sdk.server.exceptions.ServerResponseException: Session
is already recording

Which seem a bit contradictory to me. Also, is it possible to work
with multiple test id's for unit/performance/functional tests?
Should the tag be repeated per parralel node or be in the root of the file?

Best regards,

Harrie

**Comment by Harriebo** [comment]
I will make a better example of our jenkins file, now it connects in the admin I still get errors and I see maybe my questions are not that clear.
Our build file structure:

```
stage("Build"){
     appMonBuildEnvironment(systemProfile: 'Jenkins', recordSession: true) {
          dynatraceid = appMonRegisterTestRun category: 'performance'
          dynatraceiduidriven = appMonRegisterTestRun category: 'uidriven'
          //First id will be passed to unit and integration tests, second to functional tests
         // at this point it already states: Registering test run via Dynatrace Server REST interface...
ERROR: Dynatrace AppMon Plugin - build step execution failed (see the stacktrace to get more information):
com.dynatrace.sdk.server.exceptions.ServerResponseException: No connection to the Performance Warehouse. Please establish a connection from the Dynatrace Server to the Performance Warehouse to see data in this view.


          //next parameters are passed to build and test commands by maven (this part already worked without the pipeline plugin, test where visisble in dynatrace, we used the maven plugin to register the test before.

           mavenopts += "-agentpath:/opt/dynatrace/agent/lib64/libdtagent.so=name=Maven_Jenkins,server=collectionn.server:9998,optionTestRunIdJava=${dynatraceid}"
          mavenoptsuidriven += "-agentpath:/opt/dynatrace/agent/lib64/libdtagent.so=name=Maven_Jenkins,server=collectionn.server:9998,optionTestRunIdJava=${dynatraceiduidriven}"
          dynatraceoptions += "-Dtest.dynatrace=true -DdtTestrunID=${dynatraceid} -DdtTestrunIDUI=${dynatraceiduidriven}"


        //build and unit test here, this is already not working.
     }
}

stage("test"){
    parallel(node(){
              appMonBuildEnvironment(systemProfile: 'Jenkins', recordSession: true) {
                      //run integration tests with performance id
              }
       },node(){
              appMonBuildEnvironment(systemProfile: 'Jenkins', recordSession: true) {
                      //run functional tests 1 with uidriven id
              }
       },node(){
              appMonBuildEnvironment(systemProfile: 'Jenkins', recordSession: true) {
                      //run functional tests 2 with uidriven id
              }
       }
    )
}
```

Is this setup ok? And how can I debug why it won't connect to the profile to request the id?

**Comment by dglugla** [comment]
Seems like you lost your PWH connection. Please launch Dynatrace Client (RCP) and check if your PWH settings are correct (Settings -> Dynatrace Server -> Performance Warehouse).
Regarding your setup, TBH we haven't tried the plugin this way. Usually TestRun is recorded right before it's used, and I'm not sure if the TestRun ids will be passed correctly. In your setup I see that you want to use the same TestRun for UI-Driven tests executed on 2nd and 3rd node. That's also not an usual configuration, especially that you record separate sessions for each node/step. We recommend to register TestRun after session recording has started and finish it before stopping the recording. This way whole TestRun fits into a session.

Let us know if anything changed after connecting the PWH.

BR

**Comment by Harriebo** [comment]
Ah, you are right. It seemd a other engeneer was updating our dynatrace
setup while I was testing. :) sorry for that.
I got it working fine, also over mutiple nodes with multiple sets. But I
had to change the setup. The appMon tag is now in the root of the Jenkins
file and all node calls are inside of that, Calling multiple builds / test
cycles seems not to be a problem and it records fine. I did remove the
functional test because peformance testing those did not make a lot of
sense.
Thank you for the support.
